diff --git a/node_modules/@omnigraph/openapi/cjs/getJSONSchemaOptionsFromOpenAPIOptions.js b/node_modules/@omnigraph/openapi/cjs/getJSONSchemaOptionsFromOpenAPIOptions.js
index 6b0fa1b..83e7d0b 100644
--- a/node_modules/@omnigraph/openapi/cjs/getJSONSchemaOptionsFromOpenAPIOptions.js
+++ b/node_modules/@omnigraph/openapi/cjs/getJSONSchemaOptionsFromOpenAPIOptions.js
@@ -240,7 +240,8 @@ async function getJSONSchemaOptionsFromOpenAPIOptions(name, { source, fallbackFo
             if ('requestBody' in methodObj) {
                 const requestBodyObj = methodObj.requestBody;
                 if ('content' in requestBodyObj) {
-                    const contentKey = Object.keys(requestBodyObj.content)[0];
+                    const contentKeys = Object.keys(requestBodyObj.content);
+                    const contentKey = contentKeys.find(contentKey => typeof contentKey === 'string' && contentKey.match('json')) || contentKeys[0];
                     const contentSchema = requestBodyObj.content[contentKey]?.schema;
                     if (contentSchema && Object.keys(contentSchema).length > 0) {
                         operationConfig.requestSchema = contentSchema;
diff --git a/node_modules/@omnigraph/openapi/esm/getJSONSchemaOptionsFromOpenAPIOptions.js b/node_modules/@omnigraph/openapi/esm/getJSONSchemaOptionsFromOpenAPIOptions.js
index ab81247..0897139 100644
--- a/node_modules/@omnigraph/openapi/esm/getJSONSchemaOptionsFromOpenAPIOptions.js
+++ b/node_modules/@omnigraph/openapi/esm/getJSONSchemaOptionsFromOpenAPIOptions.js
@@ -237,7 +237,8 @@ export async function getJSONSchemaOptionsFromOpenAPIOptions(name, { source, fal
             if ('requestBody' in methodObj) {
                 const requestBodyObj = methodObj.requestBody;
                 if ('content' in requestBodyObj) {
-                    const contentKey = Object.keys(requestBodyObj.content)[0];
+                    const contentKeys = Object.keys(requestBodyObj.content);
+                    const contentKey = contentKeys.find(contentKey => typeof contentKey === 'string' && contentKey.match('json')) || contentKeys[0];
                     const contentSchema = requestBodyObj.content[contentKey]?.schema;
                     if (contentSchema && Object.keys(contentSchema).length > 0) {
                         operationConfig.requestSchema = contentSchema;
