diff --git a/node_modules/json-machete/cjs/healJSONSchema.js b/node_modules/json-machete/cjs/healJSONSchema.js
index 030aea2..36ffc5a 100644
--- a/node_modules/json-machete/cjs/healJSONSchema.js
+++ b/node_modules/json-machete/cjs/healJSONSchema.js
@@ -98,6 +98,20 @@ async function healJSONSchema(schema, { logger = new utils_1.DefaultLogger('heal
                     subSchema.oneOf.push(...subSchema.allOf[0].oneOf);
                     delete subSchema.allOf;
                 }
+                // Required is inherited with swagger, but not when generating the Graphql types
+                if (subSchema.required?.length && subSchema.allOf?.length) {
+                    logger.debug(`${path} has a required definition and allOf definition. Applying required to sub schemas.`);
+                    subSchema.allOf = subSchema.allOf.map((schema) => {
+                        if (schema.$resolvedRef) {
+                            return schema;
+                        }
+                        if (!schema.required) {
+                            schema.required = [];
+                        }
+                        schema.required = schema.required.concat(subSchema.required);
+                        return schema;
+                    });
+                }
                 // If they have title, it makes sense to keep them to reflect the schema in a better way
                 if (!subSchema.title) {
                     if (subSchema.anyOf != null &&
diff --git a/node_modules/json-machete/esm/healJSONSchema.js b/node_modules/json-machete/esm/healJSONSchema.js
index 9f1629f..44bd535 100644
--- a/node_modules/json-machete/esm/healJSONSchema.js
+++ b/node_modules/json-machete/esm/healJSONSchema.js
@@ -94,6 +94,20 @@ export async function healJSONSchema(schema, { logger = new DefaultLogger('healJ
                     subSchema.oneOf.push(...subSchema.allOf[0].oneOf);
                     delete subSchema.allOf;
                 }
+                // Required is inherited with swagger, but not when generating the Graphql types
+                if (subSchema.required?.length && subSchema.allOf?.length) {
+                    logger.debug(`${path} has a required definition and allOf definition. Applying required to sub schemas.`);
+                    subSchema.allOf = subSchema.allOf.map((schema) => {
+                        if (schema.$resolvedRef) {
+                            return schema;
+                        }
+                        if (!schema.required) {
+                            schema.required = [];
+                        }
+                        schema.required = schema.required.concat(subSchema.required);
+                        return schema;
+                    });
+                }
                 // If they have title, it makes sense to keep them to reflect the schema in a better way
                 if (!subSchema.title) {
                     if (subSchema.anyOf != null &&
